### ISSUE TITLE
Holding quantity not correctly calculated after split

### ISSUE THREAD

**Issue by varuk** [QuantConnect/Lean#5765]
#### Expected Behavior

When a split event of split factor 0.1428572(1:7) for a Holding quantity of 100 shares after split = 100/0.1428572 = 699.99 = 700
  (when data is in Raw mode)                                                                                                                
#### Actual Behavior

When a split event of split factor 0.1428572(1:7) for a Holding quantity of 100 shares after split = 100/0.1428572 = 699.99 = 699
(when data is in Raw mode)
#### Reproducing the Problem

https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_54573f6c83beba584fdd858718664b4e.html
#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

**Comment by Martin-Molinero** [comment]
Hey @varuk!
Thanks for the report. Lean is modeling any rounding precision issue by adding the amount as cash see [SecurityPortfolioManager.cs#L718 ](https://github.com/QuantConnect/Lean/blob/master/Common/Securities/SecurityPortfolioManager.cs#L718 ) it can be seen in the algorithms logs that there is a cash increase although **it's not** using the right price, should be adding around ~$94 but it's adding ~$644, since quantity is already scaled.
![image](https://user-images.githubusercontent.com/18473240/126016728-9c57d44d-4826-4ce4-b9a5-117a6bc4997b.png)


**TLDR: We will revisit the modeling and fix it accordingly 👍**
